pennylane/tape/tape.py

```diff
@@ -459,60 +446,14 @@ def _process_queue(self):
         self._prep = []
         self._ops = []
         self._measurements = []
-        self._output_dim = 0
 
         for obj, info in self._queue.items():
 
-            if isinstance(obj, QuantumTape):
-                self._ops.append(obj)
-
-            elif isinstance(obj, qml.operation.Operation) and not info.get("owner", False):
-                # operation objects with no owners
-
-                if self._measurements:
-                    raise ValueError(
-                        f"Quantum operation {obj} must occur prior to any measurements."
-                    )
```

Comment:
Don't we loose this logic? If so, what are the implications?

Reply:
I think this is a trade-off between efficiency and user hand-holding.

If we move toward eliminating the distinction between the three lists (`_prep`, `_ops`, and `_measurements`) in the future, we will lose these checks anyway. 

I personally think it won't cause too many problems if we get rid of these checks.

Reply:
Curious about this as well? Seems that we lose some validation, as well as logic associated with measurements.

Reply:
@mariaschuld  @josh146  I can write the validation back in somehow if we really want it. 

The other logic associated with measurements is moved to its own function, `_update_output_dim` which only loops over measurements.

Reply:
@albi3ro I'm happy as long as there is no regression in behavour/logic!

Reply:
Nice solution!